routingpy/routers/valhalla.py

```diff
@@ -192,9 +192,44 @@ def directions(
         :rtype: :class:`routingpy.direction.Direction`
         """
 
+        params = self._get_direction_params(
+            locations,
+            profile,
+            preference,
+            options,
+            units,
+            language,
+            directions_type,
+            avoid_locations,
+            avoid_polygons,
+            date_time,
+            id,
+        )
+
+        get_params = {"access_token": self.api_key} if self.api_key else {}
+
+        return self._parse_direction_json(
+            self.client._request("/route", get_params=get_params, post_params=params, dry_run=dry_run),
+            units,
+        )
+
+    @staticmethod
+    def _get_direction_params(
+        locations,
+        profile,
+        preference=None,
+        options=None,
+        units=None,
+        language=None,
+        directions_type=None,
+        avoid_locations=None,
+        avoid_polygons=None,
+        date_time=None,
+        id=None,
+    ):
         params = dict(costing=profile)
 
-        params["locations"] = self._build_locations(locations)
+        params["locations"] = Valhalla._build_locations(locations)
```

Comment:
minor thing, but shouldn't this also be a public method then?

Reply:
right, the _get_parameter_xx() shouldn't be marked as internal.. the _build_locations I had to make static now since it's called from within those new _get_param methods which are also static and they're only ever called internally, so they can stay like that I think